### PR TITLE
make RunnableGraph.addAttributes return a RunnableGraph

### LINKED ISSUE #20630: RunnableGraph.addAttributes should return same type as RunnableGraph.withAttributes
The issue can be seen by looking at the API docs:

http://doc.akka.io/api/akka/2.4.6/?_ga=1.198496171.504920134.1442952792#akka.stream.scaladsl.RunnableGraph

### PR DESCRIPTION
**Pull request by johanandren**
Supersedes #20631
Fixes #20630

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 1]
```diff
@@ -1942,4 +1936,12 @@ abstract class RunnableGraph[+Mat] extends Graph[ClosedShape, Mat] {
    * Transform only the materialized value of this RunnableGraph, leaving all other properties as they were.
    */
   def mapMaterializedValue[Mat2](f: function.Function[Mat, Mat2]): RunnableGraph[Mat2]
+
+  override def withAttributes(attr: Attributes): RunnableGraph[Mat]
+
+  override def addAttributes(attr: Attributes): RunnableGraph[Mat] =
+    withAttributes(module.attributes and attr)
+
+  override def named(name: String): RunnableGraph[Mat] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: hm, shouldn't named be addAttributes actually? Now a `named` would remove all other attributes, such as dispatchers etc

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 2]
```diff
@@ -350,6 +350,9 @@ final case class RunnableGraph[+Mat](val module: StreamLayout.Module) extends Gr
    */
   def run()(implicit materializer: Materializer): Mat = materializer.materialize(this)
 
+  override def addAttributes(attr: Attributes): RunnableGraph[Mat] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: good

**Comment by ktoso** [comment]
I have one worry here: https://github.com/akka/akka/pull/21066#discussion_r72798084
Do we have a bug with named everywhere or just here?

**Comment by johanandren** [comment]
Looks like a bug to me, It's in a few more places I can see. I'll make a separate ticket and fix those as they aren't really related to this ticket.

**Comment by ktoso** [comment]
Good. LGTM. Thanks @timcharper !

**Pull request merged**